app/src/main/java/com/example/hris/ui/profile/ProfileFragment.kt

```diff
@@ -32,29 +32,13 @@ class ProfileFragment : Fragment() {
         super.onViewCreated(view, savedInstanceState)
 
         viewModel.userData.observe(viewLifecycleOwner) {
+            val userProfile = viewModel.getProfile()
```

Comment:
This is a little bit redundant. Inside this `.observe` block, you already have access to `UserData` that can be turned into a `Profile` object. Just create a LiveData that contains the `Profile` object and observe that instead.